addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -1924,22 +1939,43 @@ export class PosStore extends WithLazyGetterTrap {
                     await this.editPartner(partner);
                 }
             }
-
             if (preset.identification === "name" && !order.floating_order_name && !order.table_id) {
                 order.floating_order_name = order.getPartner()?.name;
                 if (!order.floating_order_name) {
-                    this.editFloatingOrderName(order);
+                    await this.editFloatingOrderName(order);
+                    //re-set the order in case an order was selected from the current orders list in the EditOrderNamePopup
+                    order = this.getOrder();
+                    if (!order.floating_order_name) {
+                        return;
+                    }
                 }
             }
-
+            order.setPreset(preset);
             if (preset.use_timing && !order.preset_time) {
-                await this.syncPresetSlotAvaibility(preset);
-                order.preset_time = preset.nextSlot?.datetime || false;
+                await this.openPresetTiming(order);
+                if (!order.preset_time) {
+                    await this.syncPresetSlotAvaibility(preset);
+                    order.preset_time = preset.nextSlot?.datetime || false;
+                }
             } else if (!preset.use_timing) {
                 order.preset_time = false;
             }
         }
     }
+    async openPresetTiming(order = this.getOrder()) {
+        const data = await makeAwaitable(this.dialog, PresetSlotsPopup);
+        if (data) {
+            if (order.preset_id.id != data.presetId) {
+                await this.selectPreset(this.models["pos.preset"].get(data.presetId));
+            }
+
+            order.preset_time = data.slot.datetime;
+            if (data.slot.datetime > DateTime.now()) {
+                this.addPendingOrder([order.id]);
+                await this.syncAllOrders();
```

Comment:
Maybe only sync the order you wanna sync here
```suggestion
                await this.syncAllOrders({order: [order]});
```